client/app/components/empty-state/EmptyState.jsx

```diff
@@ -18,29 +18,30 @@ function createDashboard() {
 }
 
 function Step({ show, completed, text, url, urlText, onClick }) {
-  return show ? (
+  if (!show) {
+    return null;
+  }
+
+  return (
     <li className={classNames({ done: completed })}>
-      {onClick && urlText && <a href="javascript:void(0)" onClick={onClick}>{urlText}</a>}
-      {url && urlText && <a href={url}>{urlText}</a>}
-      {text && urlText ? ' ' : ''}
+      <a href={url || 'javascript:void(0)'} onClick={onClick}>
+        {urlText}
+      </a>{' '}
       {text}
     </li>
-  ) : null;
+  );
```

Comment:
I switched to a version of @ranbena's suggested implementation. I agree that having an early return statement when not showing is better than the ternary operator. I know I mentioned LOC in our discussions, but I still prefer readable code over short code :)

Also, in one of the cases (data source step for non admins) we render an empty <a> tag. I think it's a reasonable price to pay to keep the logic here simple.